packages/flame/lib/src/experimental/layout_component.dart

```diff
@@ -0,0 +1,146 @@
+import 'package:flame/components.dart';
+import 'package:flame/experimental.dart';
+import 'package:flame/src/game/notifying_vector2.dart';
+import 'package:flutter/widgets.dart';
+
+enum Direction { horizontal, vertical }
+
+/// Super class for [RowComponent] and [ColumnComponent]
+/// A relayout is performed when
+///  - a change in this component's children takes place
+///  - the size of this component or a parent changes
+///  - the [gap] parameter is changed
+abstract class LayoutComponent extends PositionComponent {
+  LayoutComponent(
+    this.direction,
+    this.mainAxisAlignment,
+    this._gap,
+    Vector2? size,
+  )   : isManuallySized = size != null,
+        super(
+          size: size,
```

Comment:
The `size` of the component must be set even if the user didn't specify it explicitly. Without that, the children components wouldn't be able to receive tap events.

Reply:
Passing that will be the same as not passing anything, it already defaults to the zero vector.
I think that the `size` should just be set in `layoutChildren`.

Reply:
This results in a stackoverflow error since there is a listener on size which calls `layoutChildren`.

Reply:
The listener ought to be on parent's size, whereas here we'd be setting the component's own size.
Note that the size doesn't have to be set in the constructor - we can defer setting it to the layout stage

Reply:
But if we don't have a listener on the layout component's `size` as well there won't be a call to `layoutChildren` on resizing it.